joblib/numpy_pickle.py

```diff
@@ -506,37 +547,19 @@ def load(filename, mmap_mode=None):
     """
     if Path is not None and isinstance(filename, Path):
         filename = str(filename)
-    with open(filename, 'rb') as f:
-        with _read_fileobject(f, filename, mmap_mode) as f:
-            if isinstance(f, _basestring):
-                # if the returned file object is a string, this means we try
-                # to load a pickle file generated with an version of Joblib
-                # so we load it with joblib compatibility function.
-                return load_compatibility(f)
-
-            # We are careful to open the file handle early and keep it open to
-            # avoid race-conditions on renames.
-            # That said, if data is stored in companion files, which can be
-            # the case with the old persistence format, moving the directory
-            # will create a race when joblib tries to access the companion
-            # files.
-            unpickler = NumpyUnpickler(filename, f, mmap_mode=mmap_mode)
-            try:
-                obj = unpickler.load()
-                if unpickler.compat_mode:
-                    warnings.warn("The file '%s' has been generated with a "
-                                  "joblib version less than 0.10. "
-                                  "Please regenerate this pickle file."
-                                  % filename,
-                                  DeprecationWarning, stacklevel=2)
-            except UnicodeDecodeError as exc:
-                # More user-friendly error message
-                if PY3_OR_LATER:
-                    new_exc = ValueError(
-                        'You may be trying to read with '
-                        'python 3 a joblib pickle generated with python 2. '
-                        'This feature is not supported by joblib.')
-                    new_exc.__cause__ = exc
-                    raise new_exc
+
+    if hasattr(filename, "read") and hasattr(filename, "write"):
```

Comment:
We don't need the `"write"` to `load` therefore testing only for "read" (and "seek" too).